### PR TITLE
Rename 'application-engine' and 'TAE' naming references with 'tanzu'

### PR DESCRIPTION
**Pull request by prkalle**
### What this PR does / why we need it
This PR renames 'application-engine' and TAE' naming references with 'tanzu'

Changes Summary:
- Renamed 'tae' package and API names by replacing 'TAE' with 'tanzu' appropriately
- Renamed the 'application-engine' context type with 'tanzu'
- Renamed the application-engine context metadata with names referring to 'tae' with 'tanzu' (e.g. taeOrgID with tanzuOrgID)
### Which issue(s) this PR fixes

Fixes #

### Describe testing done for PR
Unit tests and CI checks passed

### Release note

```release-note
Rename 'application-engine' and 'TAE' naming references with 'tanzu'
```

### CODE REVIEW

File: tanzu/tanzu.go [hunk 1]
```diff
@@ -1,8 +1,8 @@
 // Copyright 2023 VMware, Inc. All Rights Reserved.
 // SPDX-License-Identifier: Apache-2.0
 
-// Package tae provides APIs specific to Tanzu Application Engine(TAE)
-package tae
+// Package tanzu provides APIs specific to Tanzu(UCP)
```
Comment (prkalle) [inline tanzu/tanzu.go:4]: Would like to hear suggestions/feedback renaming the package with `tanzu`

Comment (prkalle) [reply tanzu/tanzu.go:4]: After internal discussion, moved the stuff in this package to `config` package

File: config/tanzu_context.go [hunk 2]
```diff
@@ -33,7 +31,7 @@ const (
 	customCommandName string = "_custom_command"
 )
 
-// ResourceInfo resource information of the application engine
+// ResourceInfo is the information of the tanzu resource
```
Comment (vuil) [inline config/tanzu_context.go:34]:
```suggestion
// ResourceInfo contains information identifying the Tanzu resource associated with the Context
```

File: config/tanzu_context.go [hunk 3]
```diff
@@ -101,34 +99,34 @@ func runCommand(commandPath string, args []string, opts *cmdOptions) (bytes.Buff
 	return stdout, stderr, command.Run()
 }
 
-// GetKubeconfigForContext returns the kubeconfig for any arbitrary TAE resource in the TAE object hierarchy
-// referred by the TAE context
+// GetKubeconfigForContext returns the kubeconfig for any arbitrary tanzu resource in the tanzu object hierarchy
```
Comment (vuil) [inline config/tanzu_context.go:102]: and I suggest we apply the uppercase T (for Tanzu) consistently in comments/user-facing output. From now I will call out if I see any exception to this suggestion.

File: config/contexts_test.go [hunk 4]
```diff
@@ -516,8 +519,8 @@ func TestGetContext(t *testing.T) {
 			ctxName: "test-tmc",
 		},
 		{
-			name:    "success tae",
-			ctxName: "test-tae",
+			name:    "success tanzu",
```
Comment (vuil) [inline config/contexts_test.go:522]: No need to have uppercase T

File: config/contexts_test.go [hunk 5]
```diff
@@ -619,8 +622,8 @@ func TestContextExists(t *testing.T) {
 			ok:      true,
 		},
 		{
-			name:    "success tae",
-			ctxName: "test-tae",
+			name:    "success tanzu",
+			ctxName: "test-tanzu",
```
Comment (vuil) [inline config/contexts_test.go:626]: these do not need to have uppercase T

File: config/contexts_test.go [hunk 6]
```diff
@@ -470,16 +470,19 @@ func setupForGetContext() error {
 				},
 			},
 			{
-				Name:        "test-tae",
-				ContextType: configtypes.ContextTypeTAE,
+				Name:        "test-tanzu",
```
Comment (vuil) [inline config/contexts_test.go:473]: No need to have uppercase

File: config/types/clientconfig.go [hunk 7]
```diff
@@ -22,11 +22,10 @@ const (
 	ContextTypeTMC ContextType = "mission-control"
 	contextTypeTMC ContextType = "tmc"
 
-	// ContextTypeTAE is a used to indicate the type of Context used to interact with
-	// Tanzu Application Engine (Unified Control Plane) endpoint
+	// ContextTypeTanzu represents the type of context used to interact with
+	// Tanzu (Unified Control Plane) endpoint
```
Comment (vuil) [inline config/types/clientconfig.go:26]:
```suggestion
	// Tanzu control plane endpoint
```

File: config/types/clientconfig.go [hunk 8]
```diff
@@ -179,7 +178,7 @@ func (c *ClientConfig) GetActiveContext(context ContextType) (*Context, error) {
 // GetAllCurrentContextsMap returns all current context per Target
 //
 // Deprecated: GetAllCurrentContextsMap is deprecated. Use GetAllActiveContextsMap instead
-// Note: This function will not return newly added ContextType `application-engine` information
+// Note: This function will not return newly added ContextType `tanzu` information
```
Comment (vuil) [inline config/types/clientconfig.go:181]: remove "newly added"
suggestion: ... will not return information for `tanzu` ContextType

File: config/types/clientconfig_types.go [hunk 9]
```diff
@@ -52,7 +52,7 @@ type Server struct {
 // Context configuration for a control plane. This can one of the following,
 // 1. Kubernetes Cluster
 // 2. Tanzu Mission Control endpoint
-// 3. Tanzu Application Engine endpoint
+// 3. Tanzu (UCP) endpoint
```
Comment (vuil) [inline config/types/clientconfig_types.go:55]:
```suggestion
// 3. Tanzu control plane endpoint
```

File: test/compatibility/framework/types/cfg_options.go [hunk 10]
```diff
@@ -50,8 +50,8 @@ const (
 	// ContextTypeTMC is a Tanzu Mission Control type of context.
 	ContextTypeTMC ContextType = "misson-control"
 
-	// ContextTypeTAE is a Tanzu Application Engine type of context.
-	ContextTypeTAE ContextType = "application-engine"
+	// ContextTypeTanzu is a Tanzu (UCP) type of context.
```
Comment (vuil) [inline test/compatibility/framework/types/cfg_options.go:53]:
```suggestion
	// ContextTypeTanzu is a Tanzu control plane type of context.
```

File: config/internal/kubeconfig/kubeconfig_test.go [hunk 11]
```diff
@@ -21,24 +21,24 @@ func copyFile(t *testing.T, sourceFile, destFile string) {
 }
 
 func TestReadAndMinifyKubeConfig(t *testing.T) {
-	expectedMytaeKubeconfig := `
+	expectedMytanzuKubeconfig := `
```
[the remark this answers is not in the document]
Comment (prkalle) [reply config/internal/kubeconfig/kubeconfig_test.go]: Updated. Thanks

File: config/tanzu_context.go (hunk 2 above)
Comment (prkalle) [reply config/tanzu_context.go]: updated

File: config/tanzu_context.go (hunk 3 above)
Comment (prkalle) [reply config/tanzu_context.go]: Updated all the instances of tanzu with uppercase in comments/user-facing-output(barring the suggested exceptions).Thanks!

**Review by anujc25 — COMMENTED** [review]
Thanks for the update. LGTM. I will wait for @vuil to approve the PR.

**Review by vuil — APPROVED** [review]
lgtm, thanks!

**Pull request merged**